### ISSUE TITLE
Unsure if it makes sense to warn about cast to capability from smaller int

### ISSUE THREAD

**Issue by brooksdavis** [CTSRD-CHERI/llvm#53]
It's unclear to me if we should be warning about this.  Outside the gslice slab allocator, the uses in glib seem to be storing a value in a pointer.

```
/home/bed22/git/cheriglib/lib/glib-2.0/gobject/../../../contrib/glib-2.0/gobject/gclosure.c:1113:54: warning: 
      cast to 'gpointer' (aka 'void *') from smaller integer type 'GType'
      (aka 'unsigned long') [-Wint-to-void-pointer-cast]
  closure = g_closure_new_simple (sizeof (GClosure), (gpointer) itype);
```
/cc @rwatson

**Comment by davidchisnall** [comment]
Clang issue.  We now only warn on casts from an int that's smaller than the range of a capability (i.e. 64 bits).
